beamer/health/check.py

```diff
@@ -0,0 +1,526 @@
+
+
+def get_block(chain_id: int, block_number: int) -> BlockData:
+    rpc = get_config()["rpcs"][chain_id][0]
+
+    web3 = make_web3(rpc)
+
+    return web3.eth.get_block(block_number)
+
+
+def link_to_explorer(chain_id: int, tx_hash: str) -> str:
+    return f"{get_config()['explorers'][chain_id]}{tx_hash}"
+
+
+def analyze_transfer(transfer: Transfer, ctx: Context) -> None:
+    check_if_request_created(transfer, ctx)
+    check_if_request_has_fill(transfer, ctx)
+    check_if_claim_is_made_for_fill(transfer, ctx)
+    check_if_challenge_game(transfer, ctx)
+    check_if_request_has_been_withdrawn(transfer, ctx)
+
+
+def check_if_request_created(transfer: Transfer, ctx: Context) -> None:
+    if "created" in transfer:
+        ctx.stats.requests += 1
+
+
+def check_if_request_has_fill(transfer: Transfer, ctx: Context) -> None:
+    if "created" not in transfer:
+        return
+
+    if "filled" in transfer:
+        ctx.stats.fills += 1
+        ctx.stats.volume += transfer["created"].amount
+    else:
+        if datetime.fromtimestamp(transfer["created"].valid_until) < datetime.now():
+            request = transfer["created"]
+            ctx.stats.expired_requests += 1
+
+            if not ctx.notification_state.is_set(
+                request.request_id.hex(), MessageTypes.REQUEST_EXPIRED
+            ):
+                ctx.add_message(create_expired_request_message(request))
+
+
+def check_if_claim_is_made_for_fill(transfer: Transfer, ctx: Context) -> None:
+    if "created" not in transfer:
+        return
+
+    if "filled" not in transfer:
+        return
+
+    if "claimed" in transfer:
+        ctx.stats.claims += 1
+    else:
+        request = transfer["created"]
+        fill = transfer["filled"]
+
+        if ctx.agent_address and fill.filler != ctx.agent_address:
+            return
+
+        if "withdrawn" not in transfer:
+            if (
+                datetime.fromtimestamp(transfer["created"].valid_until + claim_request_extension)
+                > datetime.now()
+            ):
+                if not ctx.notification_state.is_set(
+                    request.request_id.hex(), MessageTypes.UNCLAIMED_FILL
+                ):
+                    ctx.add_message(create_unclaimed_fill_message(request, fill))
+
+
+def check_if_challenge_game(transfer: Transfer, ctx: Context) -> None:
+    if "filled" not in transfer:
+        return
+
+    if "claimed" not in transfer:
+        return
+
+    request = transfer["created"]
+    fill = transfer["filled"]
+    claim_events = transfer["claimed"]
+
+    agent_participating = False
+    for claims in claim_events.values():
+        if any(
+            ctx.agent_address in [obj.claimer.lower(), obj.last_challenger.lower()]
+            for obj in claims
+        ):
+            agent_participating = True
+
+    agent_filled = fill.filler == ctx.agent_address
+
+    if ctx.agent_address and not (agent_filled or agent_participating):
+        return
```

Comment:
Why do you check the boolean value of `ctx.agent_address` here?

Reply:
agent_address is provided through the config. If you don't provide an agent_address - which we will do in our own cron - then we will get notification no matter who the involved party is. 
But if the agent has entered an agent_address -> then they only want to see events in which they are participating.